### PR TITLE
Basic backup logic

### PR DESCRIPTION
**Pull request by Pivnoy**
Vzdump API call for backups

### CODE REVIEW

File: nodes.go [hunk 1]
```diff
@@ -276,3 +276,16 @@ func (n *Node) GetCustomCertificates(ctx context.Context) (certs *NodeCertificat
 	err = n.client.Get(ctx, fmt.Sprintf("/nodes/%s/certificates/info", n.Name), &certs)
 	return
 }
+
+func (n *Node) Vzdump(ctx context.Context, params *VirtualMachineBackupOptions) (task *Task, err error) {
+	var upid UPID
+
+	if params == nil {
```
Comment (luthermonson) [inline nodes.go:283]: API docs read like all params are optional https://pve.proxmox.com/pve-docs/api-viewer/index.html#/nodes/{node}/vzdump

you sure you dont want to allow nil params? I haven't tested this but it reads like empty post data should still create a backup with some form of defaults. maybe confirm and let's drop the check?

File: types.go [hunk 2]
```diff
@@ -1242,3 +1242,29 @@ type FirewallIPSet struct {
...
+)
+
+type VirtualMachineBackupOptions struct {
+	VMID     uint64       `json:"vmid"`
+	Storage  string       `json:"storage"`
+	Remove   int          `json:"remove,omitempty"`
+	Mode     ModeType     `json:"mode,omitempty"`
+	Compress CompressType `json:"compress,omitempty"`
+	Notes    string       `json:"notes,omitempty"`
+}
```
Comment (luthermonson) [inline types.go:1270]: I'm not really one to talk here as I've definitely not finished param types before but if you could finish the params for this one I would appreciate it as it's an important feature and we should probably do it the first time. https://pve.proxmox.com/pve-docs/api-viewer/index.html#/nodes/{node}/vzdump note

Comment (Pivnoy) [reply types.go:1270]: Yes, I will finish it

**Comment by Pivnoy** [comment]
Add all params for vzdump API call and change enum types.
Remove return error for nil check params.

**Comment by luthermonson** [comment]
This is great! Thanks for the contribution

**Pull request merged**